### PR TITLE
[Docs] procedure for extracting a self signed cert

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Documentation

### Description

Document procedure for extracting the CA created by the selfsigned provider type.

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 1]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
 
 The messaging protocols supported depends on the type of address space used. For more information about address space types, see  link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[Address space].
 
-//TODO: Add information about retrieving the CA certificate and using
-//that in the client examples.
+[id='client-getcer-{context}']
+== Retrieving the Self Signed CA Certificate
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:23]:
```suggestion
== Retrieving the self-signed CA certificate
```

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 2]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
+
+If you opted for the `selfsigned` certificate provider type in your addressspace endpoint configuration, the generated CA
+that signed the adddressspace server certificate is required to when connecting messaging client application. The
+certificate can be retrieved from the addressspace using the following procedure.
+
+WARNING: The use of selfsigned certificates is not recommended for use in production environments.
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:29]:
```suggestion
WARNING: Using a self-signed certificate in production environments is not recommended.
```

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 3]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
 
 The messaging protocols supported depends on the type of address space used. For more information about address space types, see  link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[Address space].
 
-//TODO: Add information about retrieving the CA certificate and using
-//that in the client examples.
+[id='client-getcer-{context}']
+== Retrieving the Self Signed CA Certificate
+
+If you opted for the `selfsigned` certificate provider type in your addressspace endpoint configuration, the generated CA
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:25]:
```suggestion
If you opted for the `selfsigned` certificate provider type in your `AddressSpace` endpoint configuration, the generated CA
```

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 4]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
+that signed the adddressspace server certificate is required to when connecting messaging client application. The
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:26]:
```suggestion
that signed the `AddressSpace` server certificate is required when connecting to the messaging client application. The
```

Comment (hmaclean73) [reply documentation/assemblies/assembly-connecting-applications.adoc:26]: Is the "+" symbol required in front of line 26?

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 5]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
+ifeval::["{cmdcli}" == "oc"]
+. Log in as a messaging tenant:
++
+[subs="attributes",options="nowrap"]
+----
+{cmdcli} login -u developer
+----
+endif::[]
+
+. Retrieve the CA certificate from the addressspace required
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:42]:
```suggestion
. Retrieve the CA certificate from the `AddressSpace'.
```

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 6]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
++
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:43]: Are these "++" required here?

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 7]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
+This will give you a file containing the CA certificate, in PEM format.
++
+[subs="+quotes,attributes",options="nowrap"]
+----
+{cmdcli} get addressspace _addressspace_ -n _namespace_ -o jsonpath='{.status.caCert}{"\n"}' | base64 --decode > ca.crt
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:48]:
```suggestion
{cmdcli} get addressspace _address-space_ -n _namespace_ -o jsonpath='{.status.caCert}{"\n"}' | base64 --decode > ca.crt
```

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 8]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
+----
+
+. If a PKCS12 or JKS format trust store is required, one can be generated like this:
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:51]:
```suggestion
. If a PKCS12 or JKS format trust store is required, use the following commands to generate one:
```

File: documentation/assemblies/assembly-connecting-applications.adoc (hunk 7 above)
Comment (k-wall) [reply documentation/assemblies/assembly-connecting-applications.adoc:48]: A quick search shows me we don't use `_address_space_`.  We do use `_myspace_` and have also used  `_address-space-name_`.   I think _myspace_ is clear.

File: documentation/assemblies/assembly-connecting-applications.adoc (hunk 4 above)
Comment (k-wall) [reply documentation/assemblies/assembly-connecting-applications.adoc:26]: I don't see a `+`, are you referring to https://www.gnu.org/software/diffutils/manual/diffutils.html#Detailed-Unified

File: documentation/assemblies/assembly-connecting-applications.adoc (hunk 6 above)
Comment (k-wall) [reply documentation/assemblies/assembly-connecting-applications.adoc:43]: The first plus belongs to https://www.gnu.org/software/diffutils/manual/diffutils.html#Detailed-Unified, the second is an [Ascii doc list continuation](https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#lists).

File: documentation/assemblies/assembly-connecting-applications.adoc (hunk 7 above)
Comment (k-wall) [reply documentation/assemblies/assembly-connecting-applications.adoc:48]: Which do you prefer?

Comment (hmaclean73) [reply documentation/assemblies/assembly-connecting-applications.adoc:48]: I think _myspace_ seems to be used more frequently, so I agree with you to use _myspace_.

File: documentation/assemblies/assembly-connecting-applications.adoc (hunk 6 above)
Comment (hmaclean73) [reply documentation/assemblies/assembly-connecting-applications.adoc:43]: Looks good. Thanks for the information.

File: documentation/assemblies/assembly-connecting-applications.adoc [hunk 9]
```diff
@@ -19,8 +19,54 @@ To connect to the messaging service from outside the {KubePlatform} cluster, TLS
...
 The messaging protocols supported depends on the type of address space used. For more information about address space types, see  link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[Address space].
 
-//TODO: Add information about retrieving the CA certificate and using
-//that in the client examples.
+[id='client-getcer-{context}']
+== Retrieving the self-signed CA certificate
+
+If you opted for the `selfsigned` certificate provider type in your `AddressSpace` endpoint configuration, the generated CA
+that signed the `AddressSpace` server certificate is required when connecting to the messaging client application. The
+certificate can be retrieved from the addressspace using the following procedure.
```
Comment (hmaclean73) [inline documentation/assemblies/assembly-connecting-applications.adoc:27]:
```suggestion
certificate can be retrieved from the `AddressSpace' using the following procedure.
```

**Pull request merged**